src/platform/plugins/shared/esql/ADD_COMMAND_GUIDE.md

```diff
@@ -0,0 +1,191 @@
+# Adding Support for a New ES|QL Command in Kibana
+
+## Overview
+
+Integrating a new ES|QL command into Kibana’s editor requires you to modify the codebase in several places. This guide aims to gather all the required changes in one place. For detailed explanations of the inner workings of each package, you will be linked to its README.
+
+Seamlessly integrating a new command involves:
+
+- [ ] Creating a new node in the ES|QL abstract syntax tree (AST)
+- [ ] Validating that the command works well when prettifying the query
+- [ ] Supporting the new command in the syntax highlighting libraries
+- [ ] Creating the command definition
+- [ ] Adding the corresponding client-side validations
+- [ ] Adding the autocomplete suggestions
+
+## Creating a new node in the ES|QL AST
+
+We use a custom AST as a helper to support the rest of the capabilities listed in this document. Therefore, the first step is to create a new node in the tree when parsing the new command.
+
+- [ ] Make sure that the new command is in the local Kibana grammar definition. The ANTLR lexer and parser files are updated every Monday from the source definition of the language at Elasticsearch (via a manually merged, automatically generated [PR](https://github.com/elastic/kibana/pull/213006)).
+- [ ] Create a factory for generating the new node. The new node should satisfy the `ESQLCommand<Name>` interface. If the syntax of your command cannot be decomposed only in parameters, you can hold extra information by extending the `ESQLCommand` interface. I.E., check the Rerank command.
+- [ ] While ANTLR is parsing the text query, we create our own AST by using `onExit` listeners at `kbn-esql-ast/src/parser/esql_ast_builder_listener.ts`. Implement the `onExit<COMMAND_NAME>` method based on the interface autogenerated by ANTLR and push the new node into the AST.
```

Comment:
This is true right now, but I want to remove this ANTLR listener API, it is not necessary to use it.

Reply:
We can update the docs when you remove it Vadim